.github/workflows/pages.yml

```diff
@@ -0,0 +1,48 @@
+name: Github Pages
+
+# By default, runs if you push to main. keeps your deployed app in sync with main branch.
+on:
+  push:
+    branches:
+      - main
+# to only run when you do a new github release, comment out above part and uncomment the below trigger.
+# on:
+#   release:
+#     types:
+#       - published
+
+permissions:
+  contents: write # for committing to gh-pages branch.
```

Comment:
With a `gh-pages` branch we should be able to delete the `/docs` repo

Reply:
Yes

but what the reason for using gh-pages->github pages

and not just a github actions uploading to the github pages ?


Combination of
- https://github.com/actions/upload-pages-artifact
- https://github.com/actions/deploy-pages

Reply:
I didn't know that was an alternative.

I just want something that deploys a new GitHub pages on each merge to `main`, just like on https://github.com/emilk/eframe_template and https://github.com/emilk/egui_plot

And I don't want two `index.html` copies in the same repo